---
title: "DDIG-can"
author: "TC"
date: "2017/11/15"
output:
  xaringan::moon_reader:
    css: [default]
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
```


## Background
- cancer mutations: 
  - drivers
  - passengers

- ddig:
  - ddig_in
  - ddig_fs
  - ddig_sn
  - **ddig_can** cancer specific
**A cancer driver predictor**

---
## prediction used in previous work

- 50 features  
    - DNA  
        - conservation  
            - phylop: multi-species  
            - pssm: multiple sequence alignment  
        - KAKS:   
    - proteins  
        - Secondary Structure tendency (SS)  
            - Helix, Coil, Sheet, etc.  
        - Accessible surface area(ASA)  
        - disorder tendency (D)  
    - RNA (contributed by Mark, but not used in here due to calculation errors)  

*** 

I finally tweaked several scripts to calculate all the features.

---
## Feature analysis

a test mutation set:
- 1400 missense mutations in cancer, 1:1
  - Driver: 700, recurrence >=5
  - Passenger: 700, recurrence == 1 
    - Derived from the same genes as drivers

...Finally:
- training set: 700 random sampled mutations (1:1)
- test set: the rest 700 mutations (1:1)

---
## logistic regression model
- A logistic model is fitted to the training set, then predicted with the test set
    - Accuracy: 0.600 AUC: 0.642
    
```{r message=FALSE,echo=FALSE,fig.align="center"}
library(ROCR)
load("~/GIT/ddig-can/data/logisticROC.rda")
plot(prf)
```

---
## An SVM model
- Similar performance 
    - accuracy: 0.597, AUC: 0.644
```{r message=FALSE,echo=FALSE,fig.align="center"}
library(ROCR)
load("~/GIT/ddig-can/data/SVMROC.rda")
plot(prf)
```


---
## Model analysis
- anova analysis revealed that the following features are significant ("p<0.05")
    - asa, beta-sheet tendency contributed greatest in logistic model,
    - dna conservation, js divergence(?) exon number, helix contributed second

```
                              Df Deviance Resid. Df Resid. Dev  Pr(>Chi)    
NULL                                            699     970.31              
asa_tor                        1  11.9702       678     900.79 0.0005406 ***
sheet_p_mean                   1  12.7250       671     884.12 0.0003608 ***
dna_cons_tor                   1   8.9434       698     961.37 0.0027848 ** 
js_divergence_max              1   7.2005       693     950.64 0.0072883 ** 
num_exons_min                  1   7.3814       692     943.26 0.0065903 ** 
helix_p_tor                    1   6.8471       661     865.14 0.0088783 ** 
hhm_md_tor                     1   4.9891       688     936.51 0.0255069 *  
coil_p_max                     1   6.2100       687     930.30 0.0127034 *  
asa_max                        1   4.2320       684     922.34 0.0396687 *  
pssm_cons_mean                 1   5.4384       682     916.04 0.0196984 *  
disorder_tor                   1   3.9249       665     877.71 0.0475747 *  
helix_p_max                    1   5.1881       664     872.52 0.0227425 *  
pssm_cons_tog                  1   3.9675       660     861.18 0.0463874 *  
coil_p_mean                    1   6.5244       652     852.60 0.0106403 *  
...
```

---
## Ongoing

- HTS features
- test for other mutation types with previous features.

