<!DOCTYPE html>
<html>
  <head>
    <title>SPalign for RNA</title>
    <meta charset="utf-8">
    <meta name="author" content="TC" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# SPalign for RNA
### TC
### 2018/07/11

---





&lt;style&gt;
    .remark-slide-number {
      position: inherit;
    }

    .remark-slide-number .progress-bar-container {
      position: absolute;
      bottom: 0;
      height: 6px;
      display: block;
      left: 0;
      right: 0;
    }

    .remark-slide-number .progress-bar {
      height: 100%;
      background-color: red;
    }
&lt;/style&gt;


## Problem with RNA structure alignment

--

- Mostly **web servers**:  

--
(from  [omictools](https://omictools.com/rna-structure-alignment-category))
    - ARTS (2006 Nucleic Acids Res)
    - R3D Align (2013 Nucleic Acids Res)
    - iPARTS2 (2016 Nucleic Acids Res)
    - WebSTAR3D (2016 Bioinformatics) 
    - SARA (2009 Nucleic Acids Res)
    - SupeRNAlign (2017 Nucleic Acids Res)
    - **RMalign** (2018 BioRxiv)
    
--

.footnote[Only RMalign, ONLY, and it's not published yet]
---
## Problem with RNA structure alignment

--

- Mostly **Slow**

--
    - running time for 4 pairs of 16s~23s rRNA


|rRNAs |  ARTS|LaJolla   |SARA     | R3D Align| STAR3D|
|:-----|-----:|:---------|:--------|---------:|------:|
|pair1 | 117.2|119 886.7 |27 035.2 |     751.7|    1.7|
|pair2 |  98.5|125 835.9 |26 184.8 |     573.4|    2.0|
|pair3 |  79.7|152 635.6 |27 467.3 |     653.2|    1.4|
|pair4 |  20.1|16 209.3  |4714.4   |     308.9|    1.1|

.footnote[Ge, P., &amp; Zhang, S. (2015). STAR3D: a stack-based RNA 3D structural alignment tool. Nucleic acids research, 43(20), e137-e137.
]


---
class: center, inverse, middle

## "Ok! Let's do it ourselves."
### - Peng

---

## Start from SPalignNS

- What is changed:
--
    - Added a new rna class to support RNA
--
    - Turned off Seeds initialized from secondary structure
        - 2 of 4 seeds initialization
--
    - Other minor tweaks on makefile, parameter namespace ...
    
---

## Basic Tests

.pull-left[
- 1a9n native vs 4 decoy structure(M1 ~ M500)


|Query | RMSD| Nali|    SP|    TM|   GDT|
|:-----|----:|----:|-----:|-----:|-----:|
|M1    | 0.19|   24| 0.690| 0.946| 1.000|
|M100  | 3.78|   24| 0.382| 0.119| 0.510|
|M300  | 3.47|   18| 0.308| 0.081| 0.375|
|M500  | 4.49|   19| 0.221| 0.054| 0.312|


]
.footnote[SP: SPscore, TM: TMscore, 
GDT: global distance test score]
.pull-right[

![](https://i.imgur.com/7Ej9NbB.gif)&lt;!-- --&gt;

]

---

## Result2 Running Time Test

- 16s &amp; 23s tRNA (5wnu_A 1498nt , 1ffk_0, 2706nt )

```
# time
real	1m19.111s
user	1m19.042s
sys	0m0.024s

RMSD: 4.76
SPe: 0.193


```

---
## Others

- [Dropped it on github](https://github.com/tcgriffith/spalign_rna) with MIT open source license*(the most popular one)

- aspen: `/project/aspen/TC/GIT/spalign_rna`


--

.footnote[*By the way, my blog is also licensed under [CC BY-NC-SA 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/)]

---

```
################## Alignment Report ##################
1ffk_0_23s.pdb 5wnu_A_16s.pdb: Length= 2706 1498
SP-score (SPe): 0.193 (SPa: 0.076; SPb: 0.097)
Effective_Length: 560; Pfold= 0.1%
RMSD/Nali= 4.76 / 158 ;
GDT= 0.043 ;
TMscore(a,b,c)= 0.094 0.124 0.076 SEQID= 8.7%

Rotation Matrix:
  33.93113 -0.00738  0.62619  0.77964
 178.28775  0.03860 -0.77890  0.62596
 -46.60058  0.99923  0.03471 -0.01842

Alignment: 1ffk_0_23s.pdb 5wnu_A_16s.pdb		Structure Overlap: 4.01 %
```
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
slideNumberFormat: (current, total) => `
        <div class="progress-bar-container">
          <div class="progress-bar" style="width: ${current/total*100}%">
          </div>
        </div>
      `
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script>
(function() {
  var i, text, code, codes = document.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
})();
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
