<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Work Report: Evolutional couplings RNA</title>
    <meta charset="utf-8" />
    <meta name="author" content="TC" />
    <meta name="date" content="2019-07-03" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Work Report: Evolutional couplings RNA
### TC
### 2019-07-03

---




## Summary of DCA algorithm

A sequence `\(a(a_1,a_2...a_N)\)` with the `\(a_i\)` representing aligned protein or RNA sequences, with q=20 or q=4 types of residues/nucleotides. the probability of `\(a\)`:

![](https://wikimedia.org/api/rest_v1/media/math/render/svg/0ab29952eb08598a2eaa7135860b33ec4feaf3e1)

- J, h are the parameters of the DCA model
   - `\(J_{i,j}\)`: couplings of position i and j
   - `\(h_i\)`: field i
- Z: partition func

&gt; The central point of DCA is to interpret the `\(J_{i,j}\)` as the direct couplings between positions i and j. 

As Z is difficult to calculate (sum of the probability of all possible sequences), many methods are developed to calculate the parameters 
- number of J: `\(N*(N-1)/2*q*q\)`, 
- h: `\(N*q\)`

---

## Inferences
![](https://wikimedia.org/api/rest_v1/media/math/render/svg/0ab29952eb08598a2eaa7135860b33ec4feaf3e1)

## Mean-field 
Rely on the inverse of the covariance matrix, the core idea is 
to use the inverse the covariance matrix C

`\(C_{i,j}(ai,aj) = f_{i,j}(ai,aj) -f_i(ai)*f_j(aj)\)`

`\(J_{ij}(a_i,a_j) = -(C^{-1})_{i,j}(a_i,a_j)\)`

where `\(f_i(ai)\)` and `\(f_{i,j}(ai,aj)\)` are the frequency counts

### Pseudo-likelihood
- Directly estimate the parameters, with L2 regularization to avoid overfitting.


---

## DCA in sequence analysis

Proteins contact prediction

- Mean-field
  - mfDCA (2011, PNAS)
  - PSICOV (2012, Bioinformatics)
- Pseudo-likelihood
  - GREMLIN (2013, PNAS, David Baker)
  - plmDCA (2013, Phys Rev)
      - gplmDCA (2014, PLoS computational biology)

RNA contact prediction
- Mean-field
  - mfDCA (2015, NAR)
- Pseudo-likelihood
  - plmc (called EV-couplings) (2016, cell)

---

## Project

- Comparison of current RNA contact prediction methods(mainly DCA)

## Data
- 51 RFAM families with structure


---

## Results
[SS pairs &amp; contact maps](/190702_sel1_analysis.html)
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
