---
title: "Workreport: Aligning networks"
author: "TC"
date: "2020-01-22"
output:
  xaringan::moon_reader:
    lib_dir: libs
    nature:
      chakra: "https://cdnjs.cloudflare.com/ajax/libs/remark/0.14.0/remark.min.js"
      # beforeInit: "https://api.sparks-lab.org/js/sparkscloud.js"
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,warning=FALSE,message=FALSE, cache= TRUE, fig.align="center", fig.height=5, fig.width=9)

# library(Matrix)
library(dplyr)
library(ggplot2)
library(patchwork)


library(threejs)

library(igraph)
load("~/GIT/RNA_PKALN/rdata/gglist.rda")
# load("~/GIT/rna_ml/rdata/gglist.rda")
```

## Background

- Input: a query RNA sequence 
  -> RNAcmap
     - blastn search -> MSA1
     - Infernal+predictedSS+MSA1 -> CM
     - <mark>Infernal CM search ->MSA2</mark>
        - Aligning a sequence to a CM model (generalized HMM model with tree-topology)
        - Pseudoknots are not able to be aligned.
        
## Goal
- Improve RNA->to MRF model alignment

---
## Related works

- Dynamic programming
  - map_align (Science, 2017)
     - iterative double Dynamic programming  
     - Initial "seed" is important, which might not ideal

- Network alignment
  - TOPAS (Bioinformatics, 2019)
     - Global network alignment based on topology and sequence similarity 
     - use RNAfold predicted basepair probability -> network
     
  - ISOrank (PNAS, 2008)
     - Global network alignment of Proteinâ€“protein interactions (PPIs) and their networks

---
## Network alignment algorithm based on isorank and TOPAS

.pull-left[
- Convert RNA(RNA1, RNA2) to network(G1,G2)
    - node: nucleotides
    - edges: basepair or backbone-connection

- Let R(ij) be the similarity score for the node pair (i, j) where i is from network G1 and j is from network G2.

- (i, j) is a good match if the i and j sequences align and their respective neighbors are a good match with each other.

]


.pull-right[
```{r}
load("~/GIT/RNA_PKALN/rdata/mat1mat2.graph.rda")

nodes=rbind(mat1.graph$nodes,mat2.graph$nodes)

edges=rbind(mat1.graph$edges,mat2.graph$edges)

visNetwork::visNetwork(nodes, edges, width="100%", height="450px")
```

]


---
## toy result


```{r}
gglist$onlypair+ggtitle("basepair-only")+
  gglist$onlynb+ggtitle("backbone-only")+ patchwork::plot_layout(guides="collect")

```

---
## Result


```{r}

  gglist$nbseq+ggtitle("backbone+seq")+
  gglist$bpseq+ggtitle("basepair+seq")+
  gglist$bpnb+ggtitle("basepair+backbone")+
  patchwork::plot_layout(guides="collect")

```

---
## Ongoing

- Testing on map_align data

- Testing on "gold standard RNA alignment",possibly t-RNA, compare with map-align

- check existing RNAcmap MSA

- Check RFAM MSA



---
## Detailed algorithm

- Let N(u) be the number of neighbors of node U. so we have

$$R_{ij} =\sum _{u\in N(i)} \sum_{v \in N(j)}\frac{1}{|N(u)||N(v)|}R_{uv}$$
as topology constraint

- rewrite constraint into matrix A[i,j][u,v] and R(i,j) into vector R, we have
$$R=AR$$

- R is a eigen vector of A and can be solved by power method

$$R(k+1)=AR(k)/|AR(k)|$$

---
.pull-left[
### Sequence-constraint

- Sequence need to be continuous, so the similarity R(i,j) of mapping i to j must meet the constraint: 

$$R(i,j)=\frac{1}{2}(R(i-1,j-1)+R(i+1,j+1))$$

Can be written into a constraint matrix $A_{nb}[i,j][u,v]$ 

]


.pull-right[
### Sequence similarity constraint

- Sequence similarity (B(i,j)) between (G1 and G2)
if i and j are the same nucleotide type, then B(i,j)=1, followed by normalization $E=B/|B|$ 
then use vector E to populate $A_{seq}$
$$A_{seq}=EE^{T}$$
]

---
## put constraints together

$$A=\alpha * A_{bp} + \beta * A_{nb} + (1-\alpha -\beta) * A_{seq}$$

Then solve the eigenvector of A use power method.



