<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Workreport: Aligning networks</title>
    <meta charset="utf-8" />
    <meta name="author" content="TC" />
    <meta name="date" content="2020-01-22" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
    <link href="libs/vis-4.20.1/vis.css" rel="stylesheet" />
    <script src="libs/vis-4.20.1/vis.min.js"></script>
    <script src="libs/visNetwork-binding-2.0.9/visNetwork.js"></script>
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Workreport: Aligning networks
### TC
### 2020-01-22

---




## Background

- Input: a query RNA sequence 
  -&gt; RNAcmap
     - blastn search -&gt; MSA1
     - Infernal+predictedSS+MSA1 -&gt; CM
     - &lt;mark&gt;Infernal CM search -&gt;MSA2&lt;/mark&gt;
        - Aligning a sequence to a CM model (generalized HMM model with tree-topology)
        - Pseudoknots are not able to be aligned.
        
## Goal
- Improve RNA-&gt;to MRF model alignment

---
## Related works

- Dynamic programming
  - map_align (Science, 2017)
     - iterative double Dynamic programming  
     - Initial "seed" is important, which might not ideal

- Network alignment
  - TOPAS (Bioinformatics, 2019)
     - Global network alignment based on topology and sequence similarity 
     - use RNAfold predicted basepair probability -&gt; network
     
  - ISOrank (PNAS, 2008)
     - Global network alignment of Proteinâ€“protein interactions (PPIs) and their networks

---
## Network alignment algorithm based on isorank and TOPAS

.pull-left[
- Convert RNA(RNA1, RNA2) to network(G1,G2)
    - node: nucleotides
    - edges: basepair or backbone-connection

- Let R(ij) be the similarity score for the node pair (i, j) where i is from network G1 and j is from network G2.

- (i, j) is a good match if the i and j sequences align and their respective neighbors are a good match with each other.

]


.pull-right[
<div id="htmlwidget-e58babb043cc1074e4d8" style="width:100%;height:450px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-e58babb043cc1074e4d8">{"x":{"nodes":{"id":[1,2,3,4,5,1001,1002,1003,1004,1005],"label":["a","b","c","d","e","1","2","3","4","5"],"group":["1","1","1","1","1","2","2","2","2","2"]},"edges":{"from":[1,2,3,3,1001,1002,1003,1003],"to":[2,3,4,5,1002,1003,1004,1005]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false}},"groups":["1","2"],"width":"100%","height":"450px","idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)"},"evals":[],"jsHooks":[]}</script>

]


---
## toy result


&lt;img src="200226_files/figure-html/unnamed-chunk-2-1.png" style="display: block; margin: auto;" /&gt;

---
## Result


&lt;img src="200226_files/figure-html/unnamed-chunk-3-1.png" style="display: block; margin: auto;" /&gt;

---
## Ongoing

- Testing on map_align data

- Testing on "gold standard RNA alignment",possibly t-RNA, compare with map-align

- check existing RNAcmap MSA

- Check RFAM MSA



---
## Detailed algorithm

- Let N(u) be the number of neighbors of node U. so we have

`$$R_{ij} =\sum _{u\in N(i)} \sum_{v \in N(j)}\frac{1}{|N(u)||N(v)|}R_{uv}$$`
as topology constraint

- rewrite constraint into matrix A[i,j][u,v] and R(i,j) into vector R, we have
`$$R=AR$$`

- R is a eigen vector of A and can be solved by power method

`$$R(k+1)=AR(k)/|AR(k)|$$`

---
.pull-left[
### Sequence-constraint

- Sequence need to be continuous, so the similarity R(i,j) of mapping i to j must meet the constraint: 

`$$R(i,j)=\frac{1}{2}(R(i-1,j-1)+R(i+1,j+1))$$`

Can be written into a constraint matrix `\(A_{nb}[i,j][u,v]\)` 

]


.pull-right[
### Sequence similarity constraint

- Sequence similarity (B(i,j)) between (G1 and G2)
if i and j are the same nucleotide type, then B(i,j)=1, followed by normalization `\(E=B/|B|\)` 
then use vector E to populate `\(A_{seq}\)`
`$$A_{seq}=EE^{T}$$`
]

---
## put constraints together

`$$A=\alpha * A_{bp} + \beta * A_{nb} + (1-\alpha -\beta) * A_{seq}$$`

Then solve the eigenvector of A use power method.
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"chakra": "https://cdnjs.cloudflare.com/ajax/libs/remark/0.14.0/remark.min.js",
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
